wallet-new/docs/updates.md

```diff
@@ -0,0 +1,563 @@
+      SoftwareVersion:
+        required:
+          - svAppName
+          - svNumber
+        properties:
+          svAppName:
+            $ref: '#/definitions/ApplicationName'
+          svNumber:
+            maximum: 4294967295
+            minimum: 0
+            type: integer
+        type: object
+      ApplicationName:
+        required:
+          - getApplicationName
+        properties:
+          getApplicationName:
+            type: string
+        type: object
+      BlockVersion:
+        required:
+          - bvMajor
+          - bvMinor
+          - bvAlt
+        properties:
+          bvMajor:
+            maximum: 65535
+            minimum: 0
+            type: integer
+          bvMinor:
+            maximum: 65535
+            minimum: 0
+            type: integer
+          bvAlt:
+            maximum: 255
+            minimum: 0
+            type: integer
+        type: object
+
+# Specification<a id="sec-5" name="sec-5"></a>
+
+## New update system<a id="sec-5-1" name="sec-5-1"></a>
+
+**Summary**
+
+`We propose changes to the wallet backend data model and the way the installers interact with the cardano launcher.`
+
+To begin with, we will stop downloading an update in the background, but rather we will give the user full control.
+When requested, an update will be downloaded in a transient directory and when the user is ready to apply it,
+the downloaded installer will be moved to the folder the Cardano launcher currently watches, so that the
+installation process can be reused without changes to the underlying infrastructure.
+
+We propose adding some extra fields to what used to be called `CUpdateInfo` (which here we rename for
+clarity):
+
+    data WalletSoftwareUpdate = WalletSoftwareUpdate
+        { wsuSoftwareVersion :: !SoftwareVersion
+        , wsuBlockVesion     :: !BlockVersion
+        , wsuScriptVersion   :: !ScriptVersion
+        , wsuImplicit        :: !Bool
+        , wsuVotesFor        :: !Int
+        , wsuVotesAgainst    :: !Int
+        , wsuPositiveStake   :: !CCoin
+        , wsuNegativeStake   :: !CCoin
+        , wsuSeverity        :: !UpdateSeverity
+        -- ^ The severity of the update (i.e. optional or mandatory).
+        , wsuUpdateState     :: !UpdateState
+        -- ^ The state of this update from users point of view.
+        , wsuSha256Checksum  :: !Sha256Checksum
+        -- ^ The checksum expected for this installer. It will probably
+        -- need to be passed as part of the `UpdateProposal` (maybe in an attribute?),
+        -- or we could reuse the hash which is part of the name, if possible.
+        } deriving (Eq, Show, Generic, Typeable)
+
+where `UpdateSeverity` is defined as an enumeration indicating how compulsory this update is:
+
+    data UpdateSeverity = Mandatory
+                        | Optional
+
+For hard forks updates we would set the severity to `Mandatory`, for soft forks and other non-critical updates
+we would label an update as `Optional`.
```

Comment:
Uhm, I see your point, and I don't have a definitive answer.

What I can tell you though, is that starting with just `Optional` sounds like a good compromise, we can always add more items to the enumeration later on _or_ keep `mandatory` in place but not use it.

Note (but you know this very well already, hehe) that our current update system treats all the updates as `Mandatory` anyway, because when we restart Daedalus, the thing get's automatically applied, so that's already an improvement in this area 😀

Reply:
Ps. I have found another case for `Mandatory` updates, which would be security-critical updates, for which we definitely want to force an update. What do you think?